### ISSUE TITLE
Testing of keycloak-authz-client in keycloak-client repository

### ISSUE THREAD

**Issue by mposolda** [keycloak/keycloak#30806]
This is task to make sure that keycloak-authz-client is tested in [keycloak-client repository](https://github.com/keycloak/keycloak-client). It is similar to #30804 task, which exists for admin client. And also for https://github.com/keycloak/keycloak/issues/31978, which was already done for policy-enforcer.

I am not sure if we want to "copy" some tests or just move them and/or if we can consider moving of whole authz-client to keycloak-client repository (this is to be investigated and should be done after move of policy-enforcer). For now, we can probably copy them and then we can figure later if we are then removing them from `keycloak` repository and keep them just in `keycloak-client` repository.

**Comment by rmartinc** [comment]
@mposolda I was working on these days on this issue in this branch: https://github.com/rmartinc/keycloak-client/tree/authz-test (very initial, we need to polish this a lot but it gives an idea of the job that should be done).

I moved almost all tests which are in `org.keycloak.testsuite.authz` package. There are some issues I have detected although more or less tests are working:

1. The authz-client relies in crypto packages for the login. This is something that was not detected until we are moving the sub-project out. Probably we need to add a minimal implementation of the interface in the client package. I created one but in the test package.
2. All the events checks are removed (no event provider for testing).
3. Some tests (like PolicyEvaluationTest or some methods in UserManagedPermissionServiceTest) are using the server to evaluate the policies. This needs to change the test completely to do it remotely. I had no time to change it (I just did it in PolicyEvaluationCompositeRoleTest, this part needs to understand the test and perform the same actions remotely, so it's quite time consuming).
4. Some tests are using new modifications in the json representation that fails for version 24. It's normal, we give backswards (not forwards) compatibility. Probably we need to create a generic assume for a version or greater (those tests were not executed if the version is not the minimum one).

The package `org.keycloak.testsuite.authz.admin` is not done at all. It seems to be similar but I had not enough time to check it in detail. So my summary is that this can be done with some limitations (events or other checks that relies on test providers cannot be done, we can maintain some tests in main keycloak project). Some tests are going to take some time because we need to remove internal keycloak evaluation.
